annotation/src/main/java/io/smallrye/common/annotation/Blocking.java

```diff
@@ -0,0 +1,34 @@
+package io.smallrye.common.annotation;
+
+import static java.lang.annotation.ElementType.METHOD;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation used to indicate that the annotated method is inherently blocking and so should not be
+ * executed on a <em>non-blockable</em> thread (I/O thread, event loops...).
+ * <p>
+ * Frameworks can add support for this annotation and offload the work to another thread if the current thread cannot
+ * be blocked. It's particularly useful for frameworks using a reactive execution model. Framework relying on this
+ * annotation must specify the exact behavior:
+ * <ul>
+ * <li><em>What</em> thread is considered <em>non-blockable</em>;</li>
+ * <li>On which thread is the execution offloaded;</li>
+ * <li>Whether, when the current thread can be block, the execution of the annotated method is still offloaded to
+ * another thread, or stay on the same thread;</li>
+ * <li>If the execution of the method is offloaded, whether the initial thread is restored after the method
+ * execution.</li>
```

Comment:
Nitpicking: I don't know how this works in English, but when I write bullet lists as a sentence (where the individual items end with comma or semicolon and the last one ends with a period), the items typically don't begin with an upper-case letter :-) That might be wrong -- I'd love to learn how it really should be.

Reply:
I don't know either, so I removed the uppercase.